packages/astro/src/core/session.ts

```diff
@@ -63,12 +63,21 @@ export class AstroSession<TDriver extends SessionDriverName = any> {
 		}: Exclude<ResolvedSessionConfig<TDriver>, undefined>,
 	) {
 		this.#cookies = cookies;
+		let cookieConfigObject: AstroCookieSetOptions | undefined;
 		if (typeof cookieConfig === 'object') {
-			this.#cookieConfig = cookieConfig;
-			this.#cookieName = cookieConfig.name || DEFAULT_COOKIE_NAME;
+			const { name = DEFAULT_COOKIE_NAME, ...rest } = cookieConfig;
+			this.#cookieName = name;
+			cookieConfigObject = rest;
 		} else {
 			this.#cookieName = cookieConfig || DEFAULT_COOKIE_NAME;
 		}
+		this.#cookieConfig = {
+			sameSite: 'lax',
+			secure: true,
+			path: '/',
+			...cookieConfigObject,
+			httpOnly: true,
+		};
```

Comment:
Can't we move these lines inside the `if` block at line 70? It seems we don't need a temporary `cookieConfigObject` object.

Reply:
We need the object to be define in both cases though, and if `cookieConfig` is an object we need to spread that too, and if it's a string it needs to be used as the name instead